modules/service/src/main/scala/lucuma/odb/graphql/mapping/TargetEnvironmentMapping.scala

```diff
@@ -113,23 +121,50 @@ trait TargetEnvironmentMapping[F[_]: Temporal]
     } yield env
   }
 
-  def guideEnvironmentQueryHandler: EffectHandler[F] = {
+  def guideEnvironmentsQueryHandler: EffectHandler[F] = {
     val readEnv: Env => Result[Timestamp] = _.getR[Timestamp](ObsTimeParam)
 
     val calculate: (Program.Id, Observation.Id, Timestamp) => F[Result[List[GuideService.GuideEnvironment]]] =
       (pid, oid, obsTime) =>
         services.use { implicit s =>
           s.guideService(httpClient, itcClient, commitHash, timeEstimateCalculator)
-            .getGuideEnvironment(pid, oid, obsTime)
-            .map {
-              case Left(e)  => OdbError.GuideEnvironmentError(Some(e.format)).asFailure
-              case Right(s) => s.success
-            }
+            .getGuideEnvironments(pid, oid, obsTime)
         }
 
     effectHandler(readEnv, calculate)
   }
 
+  def guideEnvironmentQueryHandler: EffectHandler[F] = {
+    val readQueryAndCursor: (Query, Cursor) => Result[GuideService.GuideEnvironmentOptions] = (query, cursor) =>
+      // Determine if the query only contains a request for the target name(s). If so, we can avoid
+      // calling gaia under some circumstances. Explore will only be requesting the name, but
+      // Navigate will want everything.
```

Comment:
So if we specify `lookupIfUndefined: true`, and it is undefined, we call GAIA and calculate the result but don't store it for future reference?  I guess maybe I would have expected it to be cached.

Reply:
Right. Nothing is cached at the moment. Probably only navigate will be looking for the full guide star information with lookupIfUndefined === true, and since it should only do this just before it is observed I'm told the short delay is not a problem. And if it only asks for the guide star once, caching wouldn't help. That said, with the new way of doing things, we could more easily cache the result.

Reply:
We could also have explore send us the entire guide star info instead of just the name. This could change frequently while the user is choosing guide stars, so maybe we could just save it as json instead of dealing with actually saving and (especially) deleting targets from the db?

Reply:
Every time they click on the `>` button it sets the value in the database or do they have to click on "Accept" or something?

What would we do if we save it as JSON and then need to update the target JSON?  Seems like we'd need to write a Scala data migration in that case.  I guess the goal of this task was just to keep up with the selection which it seems like you accomplished.  We can consider whether/what to cache at another opportunity (i.e., maybe never :smiley:) if you like.

Reply:
> Every time they click on the > button it sets the value in the database or do they have to click on "Accept" or something?

Every time they click the > or < button. Although we can do some debouncing.

> What would we do if we save it as JSON and then need to update the target JSON? Seems like we'd need to write a Scala data migration in that case

Yeah. That wouldn't be fun. However, wouldn't any change we make to the targets invalidate the generator hash and thereby invalidate the guide target?

> We can consider whether/what to cache at another opportunity (i.e., maybe never 😃) if you like.
That sounds like great plan to me.

Reply:
> However, wouldn't any change we make to the targets invalidate the generator hash and thereby invalidate the guide target?

That's true and in fact the commit hash is folded into the calculation so any change whatsoever to the code would invalidate it.  Is that going to be a problem?  If they selected a particular target and we do some unrelated update will it disappear?  Should we just keep up with the selection regardless and do validation apart from that?

Reply:
The target selection needs to be done after the observation time/duration is set. So, for the actual observation the observer will need to select the guide target immediately before the observation is observed. I'm hoping we don't do updates to production in the middle of an observing night. 

The PI or other user can make guide star selections while they "play around", but it is the observer who will make the final selection. At the moment, at least, we're not trying to preserve what the PI selected. I'm hoping we never have to.